### DISCUSSION TITLE
Is there a way to use "--stream-sorting-excludes=1080p" but if nothing match. Take the higher one available?

### DISCUSSION THREAD

**Discussion by DivanoDova** [streamlink/streamlink#3943] [Q&A] [answered]
I use a .bat file to record some website with the option:
`best --stream-sorting-excludes=">1080p"`
But sometime the website dosn't provide any 1080p or less stream so i end up in a loop of error.
Is there a way to grab a higher one if nothing else is available?

**Comment by bastimeyer** [comment, marked as answer]
There's the `best-unfiltered` stream name which you can add to your streams list, like `best,best-unfiltered`. This ensures that it always returns a stream, even if you filter out everything with `--stream-sorting-excludes`.
https://streamlink.github.io/cli.html#cmdoption-stream-sorting-excludes

**Reply by DivanoDova** [reply]
Thank you im gonna try this , but one more question does this use the closer stream quality to 1080p and not the higher one available?
Or do i need to use `worst-unfiltered`

**Reply by bastimeyer** [reply]
The resulting streams are always ordered by their stream "weight", a specific metric for ranking them, which can depend on the plugin. The `best` synonym will select the one with the highest weight of all the available streams, and `worst` the one with the lowest value. If you filter out streams with `--stream-sorting-excludes`, then those will be ignored by `best` and `worst`, but not by `best-unfiltered` and `worst-unfiltered`.

So if you want to exclude everything above "1080p" but still select the one with the highest quality while also allowing for higher quality streams when no other quality below or equal to 1080p is available, then you should use
```
streamlink "--stream-sorting-excludes=>1080p" URL best,best-unfiltered
```

**Reply by DivanoDova** [reply]
Thanks, your help is much appreciated :)
